discovery.md

```diff
@@ -876,6 +876,75 @@ Content-Type: application/json
 ]
 ```
 
+### History
+
+The discovery service MAY include itself as a Service entity that is
+discoverable.
+
+The events produced by this "self" Service describe the management requests
+made to the discovery service. The events produced MUST include the operation
+and path to which the management request was made as well as the request body
+and the response that was given.
+
+Receiving the complete set of these will allow a downstream mirror to exactly
+materialize a consistent Service collection.  A downstream mirror MAY alter
+records as appropriate for its cirucmstances.
+
+Any Discovery Endpoint Service entities MUST adhere to the following:
+```json
+{
+  "id": "{id}",
+  ... other service attributes ...
+  "events": [
+    {
+      "specversions": ["1.x-wip"],
+      "type" : "io.cloudevents.discovery.change",
+      "dataschemacontent": { // see ce_discovery.yaml#/components/schemas/change
+        "type": "object",
+        "properties": {
+          "operation": "{operation}", // POST|DELETE
+          "path": "{path}",           // /services[/{id}]
+          "request": {request},       // as per path specification
+          "response": {response}      // as per path specification
+        }
+      }
+    }
+  ]
+}
+```
+
+#### `GET /changes`
```

Comment:
We may want to remove this and the OpenAPI changes the reflect it.  A potentially preferable alternative would allow subscribers to declare an offset or other "time" (including "NOW") from which to start receiving events of the subscribed service.  By specifying a time in the past, a historical record could be retrieved using CloudEvent supporting protocols and avoiding the use of this HTTP only interface.  The guarantees of the requested/negotiated protocols and scaling/factoring elements (such as partitioning) would thereby be a part of the delivery of the historical information.  As such it would provide for consensus over replication results and avoid some of the system pitfalls that can result from some implementation choices (e.g. imagine a partitioned data store across time where query results changed to alter the order of processing change events).